quinn-udp/src/unix.rs

```diff
@@ -180,6 +180,38 @@ fn init(io: SockRef<'_>) -> io::Result<()> {
             }
         }
     }
+    #[cfg(target_os = "freebsd")]
+    {
+        if is_ipv4 {
+            let on: libc::c_int = 1;
+            let rc = unsafe {
+                libc::setsockopt(
+                    io.as_raw_fd(),
+                    libc::IPPROTO_IP,
+                    libc::IP_RECVDSTADDR,
+                    &on as *const _ as _,
+                    mem::size_of_val(&on) as _,
+                )
+            };
+            if rc == -1 {
+                return Err(io::Error::last_os_error());
+            }
+        } else {
+            let on: libc::c_int = 1;
+            let rc = unsafe {
+                libc::setsockopt(
+                    io.as_raw_fd(),
+                    libc::IPPROTO_IPV6,
+                    libc::IPV6_RECVPKTINFO,
+                    &on as *const _ as _,
+                    mem::size_of_val(&on) as _,
+                )
+            };
+            if rc == -1 {
+                return Err(io::Error::last_os_error());
+            }
```

Comment:
Could we move this down into the common `is_ipv4` case below, and delete the identical Linux `IPV6_RECVPKTINFO` call above?

Reply:
Sure.